doc/openssl_migration_guide.rst

```diff
@@ -0,0 +1,389 @@
+            return 1;
+        }
+        EVP_PKEY* privKey = PEM_read_PrivateKey(privKeyFile, NULL, NULL, NULL);
+        fclose(privKeyFile);
+
+        // String to encrypt
+        unsigned char* plaintext = (unsigned char*)"Your great-grandfather gave this watch to your granddad for good luck. Unfortunately, Dane's luck wasn't as good as his old man's.";
+        size_t plaintext_len = strlen((char*)plaintext);
+
+        // Encrypt
+        EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new(pubKey, NULL);
+        EVP_PKEY_encrypt_init(ctx);
+        EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_OAEP_PADDING);
+        EVP_PKEY_CTX_set_rsa_oaep_md(ctx, EVP_sha256());
+        size_t encrypted_len;
+        EVP_PKEY_encrypt(ctx, NULL, &encrypted_len, plaintext, plaintext_len);
+        unsigned char* encrypted = (unsigned char*)malloc(encrypted_len);
+        EVP_PKEY_encrypt(ctx, encrypted, &encrypted_len, plaintext, plaintext_len);
+
+        // Decrypt
+        EVP_PKEY_CTX *ctx2 = EVP_PKEY_CTX_new(privKey, NULL);
+        EVP_PKEY_decrypt_init(ctx2);
+        EVP_PKEY_CTX_set_rsa_padding(ctx2, RSA_PKCS1_OAEP_PADDING);
+        EVP_PKEY_CTX_set_rsa_oaep_md(ctx2, EVP_sha256());
+        size_t decrypted_len;
+        EVP_PKEY_decrypt(ctx2, NULL, &decrypted_len, encrypted, encrypted_len);
+        unsigned char* decrypted = (unsigned char*)malloc(decrypted_len + 1);
+        EVP_PKEY_decrypt(ctx2, decrypted, &decrypted_len, encrypted, encrypted_len);
+        decrypted[decrypted_len] = '\0';
+
+        // Print encrypted and decrypted strings
+        for(size_t i = 0; i < encrypted_len; i++) {
+            printf("%02X", encrypted[i]);
+        }
+        printf("\n");
+        printf("%s\n", decrypted);
+
+        // Clean up
+        EVP_PKEY_free(pubKey);
+        EVP_PKEY_free(privKey);
+        EVP_PKEY_CTX_free(ctx);
+        EVP_PKEY_CTX_free(ctx2);
+        free(encrypted);
+        free(decrypted);
+
+        return 0;
+    }
+
+This example uses OpenSSL'S EVP interface, specifically ``EVP_PKEY_encrypt()``
+and ``EVP_PKEY_decrypt()`` functions to encrypt and decrypt a string using RSA.
+The public and private keys are loaded from files.
+The ``EVP_PKEY_CTX_set_rsa_padding()`` and ``EVP_PKEY_CTX_set_rsa_oaep_md()`` functions are used
+to set the padding scheme and the hash function for RSA-OAEP.
+
+Here is the equivalent C++ code using Botan:
+
+.. literalinclude:: /../src/examples/rsa_encrypt.cpp
+   :language: cpp
+
+This example uses the ``PK_Encryptor_EME`` and ``PK_Decryptor_EME`` classes to
+:ref:`encrypt and decrypt <api_ref/pubkey:public key encryption/decryption>`.
+a message using :ref:`api_ref/pubkey:rsa`. The public and private keys are
+:ref:`loaded from files <api_ref/pubkey:serializing private keys using pkcs #8>`.
+The padding scheme and :doc:`hash function <api_ref/hash>` are passed as a string parameter.
+
+Consider the following application code to sign some data with ECDSA using OpenSSL.
```

Comment:
Let's have another headline here. (Split asym encryption and signing into two sections)

Reply:
OK